### ISSUE TITLE
upgrade: No module named pipx

### ISSUE THREAD

**Issue by wturrell** [coderholic/pyradio#256]
```
Extracting RyRadio source code...
Looking for pipx ... /opt/homebrew/opt/python@3.12/bin/python3.12: No module named pipx
Command used: python3 -m pipx --version 2>/dev/null


Error: pipx not found.
       Please install pipx and try again
```

So this is on macOS Sonoma (14.6.1) which is mostly using Homebrew (4.3.20)

pipx is installed with homebrew:

```
$ brew list pipx
/opt/homebrew/Cellar/pipx/1.7.1/bin/pipx
[...]
```

I've tried uninstalling and adding it again.

If, instead, I try and install pipx via python3 itself, I get:

```
$ python3 -m pip install pipx
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try brew install
    xyz, where xyz is the package you are trying to
    install.
```

(same output if I do ``pip3 install pipx`` - unclear if that's really just an alias..)

pipx (1.2.0) *is* there if do `pip list` rather than `pip3 list`

I don't see a separate python3 version of the formula in homebrew?

Any other suggestions?

```
$ pyradio --v
PyRadio version: 0.9.2.25
Python version: 3.11.10 (main, Sep  7 2024, 01:03:31) [Clang 15.0.0 (clang-1500.3.9.4)]
$ python --version
bash: python: command not found
$ python3 --version
Python 3.12.6
$ pipx list
venvs are in /Users/wt/.local/pipx/venvs
apps are exposed on your $PATH at /Users/wt/.local/bin
manual pages are exposed at /Users/wt/.local/share/man
   package docx2pdf 0.1.8, installed using Python 3.11.7
    - docx2pdf
   package pyradio 0.9.2.25, installed using Python 3.11.7
    - pyradio
$ which python3
/opt/homebrew/bin/python3
$ echo "$PATH"
/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/lib/ruby/gems/3.2.0/bin:/usr/local/opt/ruby/bin:./vendor/bin:/usr/local/sbin:/usr/local/bin:/opt/homebrew/bin:/usr/local/share/npm/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/Users/wt/bin:/Users/wt/ansible/bin:/Users/wt/.composer/vendor/bin:/Users/wt/.cargo/bin:/Applications/DevDesktop/tools:/Applications/Hugin/tools_mac:/Users/wt/.local/bin
$ which pipx
/opt/homebrew/bin/pipx
$ /opt/homebrew/bin/pipx --version
1.7.1
```

**Comment by s-n-g** [comment]
Ok, so pyradio was built (through pipx) for python 3.11
I presume you have upgraded python to 3.12 recently

pipx still has the references to 3.11 (in the virtual environment)

What does  `pipx list` say?

**Comment by wturrell** [comment]
> Ok, so pyradio was built (through pipx) for python 3.11 I presume you have upgraded python to 3.12 recently

Yep..

> What does `pipx list` say?

Still refers to 3.11:

```
venvs are in /Users/wt/.local/pipx/venvs
apps are exposed on your $PATH at /Users/wt/.local/bin
manual pages are exposed at /Users/wt/.local/share/man
   package docx2pdf 0.1.8, installed using Python 3.11.7
    - docx2pdf
   package pyradio 0.9.2.25, installed using Python 3.11.7
    - pyradio
```

**Comment by s-n-g** [comment]
ok, this is what I would do

1. uninstall `pipx`
2. keep a backup of **~/.local/pipx/**, for example `tar cvzf ~/mypipx.tar.gz ~/.local/pipx`
3. remove **~/.local/pipx/shared**
4. install `pipx`
5. execute `pipx reinstall-all`

I hope the last command will install both `docx2pdf` and `pyradio`
If it fails for `pyradio`, just install it from code

**Comment by wturrell** [comment]
I'm impressed :-)

```
$ pyradio --version
XDG Dirs found; enabling XDG Base compliant operation
PyRadio version: 0.9.3.11
Python version: 3.12.6 (main, Sep  6 2024, 19:03:47) [Clang 15.0.0 (clang-1500.3.9.4)]
wt@macmini-m1 ~ $ pyradio -U
XDG Dirs found; enabling XDG Base compliant operation
Released version   :  0.9.3.11
Installed version  :  0.9.3.11
Latest version already installed. Nothing to do....
```

**Comment by s-n-g** [comment]
Great!
I think we can close this one...
Feel free to reopen it or open a new issue if a problem arises
